src/main/java/org/broadinstitute/hellbender/tools/evoquer/AlleleSubsettingUtilsForJointCalling.java

```diff
@@ -0,0 +1,217 @@
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.tools.walkers.ReferenceConfidenceVariantContextMerger;
+import org.broadinstitute.hellbender.tools.walkers.annotator.AnnotationUtils;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.variant.GATKVariantContextUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class AlleleSubsettingUtilsForJointCalling {
+    private static final Logger logger = LogManager.getLogger(AlleleSubsettingUtilsForJointCalling.class);
+
+    /**
+     *
+     * @param unmergedCalls
+     * @param maxAlleles
+     * @return
+     */
+    public static List<VariantContext> subsetAlleles(List<VariantContext> unmergedCalls, int maxAlleles, Allele longestRefAllele) {
+
+        Map<Allele, Integer> alleleSpecificQuals = new HashMap<>();
+        List<VariantContext> callsWithExpandedAlleles = sumAllQualsAndExpandAlleles(unmergedCalls, alleleSpecificQuals, longestRefAllele);
+        logger.debug("unique alleles: " + alleleSpecificQuals.size() + alleleSpecificQuals.keySet());
+
+        logger.debug("delete this log message - for testing only");
+        // choose the target alleles
+        List<Allele> targetAlleles = alleleSpecificQuals.entrySet().stream()
+                // get the highest values
+                .sorted(Map.Entry.<Allele, Integer>comparingByValue().reversed()).limit(maxAlleles)
+                .map(entry -> entry.getKey()).collect(Collectors.toList());
+        logger.debug(targetAlleles);
+
+        Set<Allele> allelesToDrop = new HashSet<>(alleleSpecificQuals.keySet());
+        allelesToDrop.removeAll(targetAlleles);
+
+        if (allelesToDrop.isEmpty()) {
+            // there were not enough alleles for us to subset any
+            return callsWithExpandedAlleles;
+        }
+
+        // how do i do this without hardcoding?
+        int[] relevantAlleleIndexes = new int[] {0,2};
+
+        List<VariantContext> results = new ArrayList<>(unmergedCalls.size());
+
+        callsWithExpandedAlleles.stream().forEach(vc -> {
+            Set<Allele> vcAltAllelesSet = new HashSet<>(vc.getAlternateAlleles());
+            Set<Allele> allelesToSubset = SetUtils.intersection(vcAltAllelesSet, allelesToDrop);
+            if (allelesToSubset.isEmpty()) {
+                results.add(vc);
+            } else if (containsMeaningfulAllele(SetUtils.difference(vcAltAllelesSet, allelesToSubset))) {
+                logger.debug("subsetting alleles for sample: " + vc.getSampleNames());
+                Genotype g =  vc.getGenotype(0);
+                GenotypeBuilder gb = new GenotypeBuilder(g);
+
+                List<Allele> galleles = g.getAlleles();
+                List<Allele> updatedGAlleles = galleles.stream().map(gallele -> {
+                    if (allelesToSubset.contains(gallele)) {
+                        return Allele.NO_CALL;
```

Comment:
This probably reads better as a ternary

```
gallele -> allelesToSubset.contains(gallele) ? Allele.NO_CALL :  galllele
```

I might pull out a method though, replaceRemovedAllelesWithNoCalls.